lib/charms/lego_base_k8s/v0/lego_client.py

```diff
@@ -182,6 +185,19 @@ def _pull_certificates_from_workload(self, csr_subject: str) -> List[Union[bytes
         chain_pem = self._container.pull(path=f"{self._certs_path}{csr_subject}.crt")
         return list(chain_pem.read().split("\n\n"))
 
+    def _sync_certificates(self, event: EventBase) -> None:
+        """Goes through all the certificates relations and handles outstanding requests."""
+        self._on_config_changed(event)
+        for relation in self.model.relations[CERTIFICATES_RELATION_NAME]:
```

Comment:
If there is no `certificates` relation, `self.model.relations[CERTIFICATES_RELATION_NAME]` returns an empty list or it crashes? Let's make sure this works fine when the charm is not (yet) related to anything.

Reply:
Fixed